### PR TITLE
Add libarchive@3.7.2

### PR DESCRIPTION
**Pull request by dzbarsky**

### CODE REVIEW

**Comment by bazel-io** [comment]
Hello @bazelbuild/bcr-maintainers, modules without existing maintainers (libarchive) have been updated in this PR. Please review the changes.

File: modules/libarchive/3.7.2/MODULE.bazel [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module(
+    name = "libarchive",
+    version = "3.7.2",
+    compatibility_level = 1,
+)
+bazel_dep(name = "xz", version = "5.4.5.bcr.1")
+bazel_dep(name = "zlib", version = "1.3.1")
+bazel_dep(name = "zstd", version = "1.5.5.bcr.1")
```
Comment (meteorcloudy) [inline modules/libarchive/3.7.2/MODULE.bazel:8]: Please add a dependency on `platforms`

**Comment by dzbarsky** [comment]
@meteorcloudy Updated, PTAL!

File: modules/libarchive/3.7.2/presubmit.yml [hunk 2]
```diff
@@ -7,7 +7,6 @@ matrix:
   - ubuntu2204
   - macos
   - macos_arm64
-  - windows
```
Comment (meteorcloudy) [inline modules/libarchive/3.7.2/presubmit.yml]: Why removing windows coverage?

Comment (meteorcloudy) [reply modules/libarchive/3.7.2/presubmit.yml]: SGTM!

**Review by meteorcloudy — APPROVED** [review]

**Pull request merged**